### PR TITLE
IPRO+MPD: Make In-Place-Resource-Optimization with MapProxyDomain work

### PR DESCRIPTION
**Pull request by oschaaf**
Fixes https://github.com/pagespeed/ngx_pagespeed/issues/1015

Pull with MPS Side for this change:
https://github.com/pagespeed/mod_pagespeed/pull/1161

### CODE REVIEW

File: src/ngx_pagespeed.cc [hunk 1]
```diff
@@ -1882,6 +1882,38 @@ ngx_int_t ps_resource_handler(ngx_http_request_t* r,
...
 
     return ps_async_wait_response(r);
+  } else if (!html_rewrite && response_category == RequestRouting::kResource) {
+    bool is_proxy = false;
+    GoogleString mapped_url;
+    GoogleString host_header;
+  
+    // TODO(oschaaf): this passed in something call stripped_url_, but at first glance
+    // that is just the original url. Double check.
+    fprintf(stderr, "pagespeed aaaaaagh is proxy check %s \n", url_string.c_str());
```
Comment (jeffkaufman) [inline src/ngx_pagespeed.cc]: remove debugging fprintf

File: src/ngx_pagespeed.cc [hunk 2]
```diff
@@ -1882,6 +1882,38 @@ ngx_int_t ps_resource_handler(ngx_http_request_t* r,
...
+    if (options->domain_lawyer()->MapOriginUrl(url, &mapped_url,
+                                         &host_header, &is_proxy) && is_proxy) {
+      fprintf(stderr, "pagespeed aaaaaagh is proxy resource determined \n");
```
Comment (jeffkaufman) [inline src/ngx_pagespeed.cc]: remove debugging fprintf

File: src/ngx_server_context.h [hunk 3]
```diff
@@ -41,7 +41,7 @@ class NgxServerContext : public SystemServerContext {
   virtual ~NgxServerContext();
 
   // We expect to use ProxyFetch with HTML.
```
Comment (jeffkaufman) [inline src/ngx_server_context.h]: comment is out of date now

File: src/ngx_pagespeed.cc [hunk 4]
```diff
@@ -1882,6 +1882,38 @@ ngx_int_t ps_resource_handler(ngx_http_request_t* r,
...
+  } else if (!html_rewrite && response_category == RequestRouting::kResource) {
+    bool is_proxy = false;
+    GoogleString mapped_url;
+    GoogleString host_header;
+  
+    // TODO(oschaaf): this passed in something call stripped_url_, but at first glance
+    // that is just the original url. Double check.
+    fprintf(stderr, "pagespeed aaaaaagh is proxy check %s \n", url_string.c_str());
+    if (options->domain_lawyer()->MapOriginUrl(url, &mapped_url,
+                                         &host_header, &is_proxy) && is_proxy) {
```
Comment (jeffkaufman) [inline src/ngx_pagespeed.cc]: weird indentation

**Comment by oschaaf** [comment]
@jeffkaufman I addressed your comments & force pushed

**Comment by jeffkaufman** [comment]
I'm seeing this failing its test:

```
TEST: Reverse proxy a pagespeed URL.
http_proxy=localhost:8051 wget --save-headers -O - http://modpagespeed.com/mod_pagespeed_example/styles/yellow.css
     check_from egrep -q [ ]*HTTP/1[.]. 200 OK
http_proxy=localhost:8051 wget --save-headers -O - http://modpagespeed.com/mod_pagespeed_example/styles/A.yellow.css.pagespeed.cf.KM5K8SbHQL.css
     check_from egrep -q [ ]*HTTP/1[.]. 200 OK
TEST: proxying from external domain should optimize images in-place.
      Fetching http://localhost:8050/modpagespeed_http/Puzzle.jpg   --save-headers -o /tmp/mod_pagespeed_test.jefftk/nginx/fetched_directory/wget_output.txt                                                   -O /tmp/mod_pagespeed_test.jefftk/nginx/fetched_directory/wget/fetch_until_output.19485 until $(grep -c PSA-aj) = 1
..
     check_from fgrep -q Etag: W/"PSA-aj-
FAILed Input: HTTP/1.1 200 OK
Content-Type: image/jpeg
Content-Length: 98051
Connection: keep-alive
Server: Apache/2.4.7 (Ubuntu)
Accept-Ranges: bytes
ETag: W/"PSA-aj-YsUBdV-i_6"
Date: Wed, 28 Oct 2015 13:23:29 GMT
Expires: Wed, 28 Oct 2015 13:33:29 GMT
Cache-Control: max-age=599
/usr/local/google/home/jefftk/modpagespeed/otto-1015/nginx/mod_pagespeed/src/pagespeed/automatic/system_test_helpers.sh:452: check_from
/usr/local/google/home/jefftk/modpagespeed/otto-1015/nginx/mod_pagespeed/src/pagespeed/system/system_test.sh:309: source
/usr/local/google/home/jefftk/ngx_pagespeed/otto-1015/ngx_pagespeed/test/nginx_system_test.sh:339: main
     failure at line 309
in 'proxying from external domain should optimize images in-place.'
FAIL.
With serf fetcher setup.
make: *** [nginx_debug_smoke_test] Error 1
```

**Comment by oschaaf** [comment]
@jeffkaufman I saw that earlier, that one was caused by comparing case insensitive in the test for `ETag` (vs `Etag`). Did I forget to include something in one of the pull requests? Double checking..

**Comment by jeffkaufman** [comment]
Whoops, I think that was my mistake.  I thought the system test change was a straight move, without changes, and imported it wrong.  Sorry!

**Comment by jeffkaufman** [comment]
I've added the `-i` flags back to the ETag fgreps and it passes now. Sorry about that!

Still waiting for the tests to run under valgrind.

**Comment by jeffkaufman** [comment]
Tests pass, merging.

**Pull request merged**